az-core/src/main/java/azkaban/utils/Props.java

```diff
@@ -743,23 +743,43 @@ public void storeFlattened(final OutputStream out) throws IOException {
   }
 
   /**
-   * Get a map of all properties by string prefix
+   * Get a flattened map of all properties by given prefix
    *
-   * @param prefix The string prefix
+   * @param prefix the prefix string
+   * @param ignoreCase if allows case sensitive checking for the given prefix string
+   * @return a flattened map with all properties with the given prefix
    */
-  public Map<String, String> getMapByPrefix(final String prefix) {
-    final Map<String, String> values = this._parent == null ? new HashMap<>() :
-        this._parent.getMapByPrefix(prefix);
+  public Map<String, String> getMapByPrefix(final String prefix, boolean ignoreCase) {
```

Comment:
the original codebase support only ALL UPPER CASE and ALL LOWER CASE prefix one. It will cause potential error if the input is a mixed case prefix. Plus, the original code will scan the list of props twice which can be optimized.